aten/src/ATen/native/cuda/UnaryOpsKernel.cu

```diff
@@ -23,6 +23,16 @@ void bitwise_not_kernel_cuda(TensorIterator& iter) {
   }
 }
 
-REGISTER_DISPATCH(bitwise_not_stub, &bitwise_not_kernel_cuda);
 
+void sign_kernel_cuda(TensorIterator& iter){
+    AT_DISPATCH_ALL_TYPES(iter.dtype(), "sign_cuda", [&]() {
```

Comment:
This would not work for all types, despite it is dispatching "ALL_TYPES": it does not handle bool and half. See `aten/src/ATen/Dispatch.h`. You might wanna handle Boolean type specifically, just like `bitwise_not` you are referring to. You can also use `AT_DISPATCH_ALL_TYPES_AND(kHalf, ...)` so half is also handled here.

Reply:
Wasn't aware that you haven't finished the bool part. No worry, take your time (:

Reply:
I almost finished Bool part, but I'm struggling with the returned tensor which doesn't share the same type as the input one. 

The current implementation I have (with casting to unsigned char) it works on CPU but CUDA throws CUDA unaligned memory error.

How should I handle this ?

Reply:
Can you push your code? The input and output types should be the same (so bool -> bool). This means that torch.sign is just the identity function on bool tensors.

Reply:
I though about this, but was not sure this was to intended behaviour.

Reply:
Just pushed bool and half support. 

I implemented bool as a no-op function over the input tensor. Let me know.